### ISSUE TITLE
use of restrictions sequence in car.lua file 

### ISSUE THREAD

**Issue by EricKiennemann** [Project-OSRM/osrm-backend#5987]
Hello, I am trying to understand the use of restrictions sequence in the setup of car.lua script. I want to know if I need to modify it and add bus / psv vehicle or not.

For this I need to understand its impact on the routing.

When I look at the profile.md file it is said : "Determines which turn restrictions will be used for this profile."
But when I look in the lua scripts I only find a use in the WayHandlers.oneway function in the way_handlers.lua file. In this function the restrictions sequence is used to check oneway way, but nothing link with "turn restrictions" as described in the profile.md file.

So at the end, I don't really understand the use of this restrictions sequence.

Someone could give me more info on this ?
Thanks

**Comment by mjjbell** [comment]
The restrictions sequence from the Lua profile is used by the C++ restriction parsing logic during graph extraction. 

https://github.com/Project-OSRM/osrm-backend/blob/9d81eb327a74745be3af28308fcc798dd1a93723/src/extractor/extractor.cpp#L433-L438

The restriction parser will include namespaced restrictions that match those in the profile sequence.

https://github.com/Project-OSRM/osrm-backend/blob/9d81eb327a74745be3af28308fcc798dd1a93723/src/extractor/restriction_parser.cpp#L77-L80

In short, you'll want to modify that sequence to include the vehicles you are interested in, and that will be reflected in the extracted routing graph.

**Comment by EricKiennemann** [comment]
Thanks for your answer : it is clear
